bokehjs/src/coffee/models/tools/gestures/select_tool.coffee

```diff
@@ -59,16 +60,7 @@ export class SelectToolView extends GestureToolView
       else
         logger.debug("Unrecognized selection geometry type: '#{g.type}'")
 
-    if final
-      tool_events = @plot_model.plot.tool_events
-      if append
-        geoms = tool_events.geometries
-        geoms.push(g)
-      else
-        geoms = [g]
-
-      tool_events.geometries = geoms
-    return null
+    @plot_model.plot.trigger_event(new Selection({geometry: g}))
```

Comment:
We might want to include `final` in the event? That would provide a way for event handlers to filter and ignore intermediate selection geometries if desired.